### PR TITLE
Add ERM section to Project Settings with appropriate permissions

### PR DESCRIPTION
**Pull request by gherceg**
## Summary

https://dimagi-dev.atlassian.net/browse/SS-173

This adds a new tabclass "Enterprise Release Management" on the Project Settings page. This tab displays if the user is an admin and the domain this `LINKED_PROJECTS` privilege enabled. A more appropriate privilege name might be `RELEASE_MANAGEMENT` as this will apply to more than just linked projects in the future so that change may be in the works. 

**This privilege will take precedence over the linked projects feature flag as the tests describe**

Screen Shot 2021-07-19 at 1 22 14 PM

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

Added automated tests to validate display logic.
### Safety story

Locally tested. Will not be directly merged to master, and will go through QA on its parent branch.
### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/tabs/tabclasses.py [hunk 1]
```diff
@@ -1974,6 +1979,22 @@ def _get_feature_flag_items(domain):
     return feature_flag_items
 
 
+def _get_release_management_items(user, domain):
+    release_management_items = []
+
+    if user.is_domain_admin(domain) and domain_has_privilege(domain, LINKED_PROJECTS):
+        release_management_items.append({
```
Comment (orangejenny) [inline corehq/tabs/tabclasses.py:1986]: These two items are on the border of deserving a helper function rather than being repeated here and above.

**Review by orangejenny — COMMENTED** [review]
This code looks good, but you also need to update `DomainLinkView` and `DomainLinkHistoryReport` to control access via the privilege/flag.

I'm ambivalent about calling this "manage releases" because we've used that to refer specifically to managing apps. I don't think that usage is very widespread - a bunch of it was CAS-related and has been removed - but we do refer to the "releases" page in user-facing docs.

File: corehq/tabs/tabclasses.py (hunk 1 above)
Comment (gherceg) [reply corehq/tabs/tabclasses.py:1986]: Agreed because I'm bound to use this elsewhere. I'll make that change.

File: corehq/apps/reports/dispatcher.py [hunk 2]
```diff
@@ -297,12 +297,18 @@ def dispatch(self, request, *args, **kwargs):
...
 
         if (report.endswith(DomainForwardingRepeatRecords.__name__)
                 and not domain_has_privilege(domain, privileges.DATA_FORWARDING)):
             return False
-        if (report.endswith(IncrementalExportLogView.__name__) and not INCREMENTAL_EXPORTS.enabled(domain)):
+        if report.endswith(IncrementalExportLogView.__name__) and not INCREMENTAL_EXPORTS.enabled(domain):
+            return False
+        if (report.endswith(DomainLinkHistoryReport.__name__)
+                and not can_access_linked_domains(request.couch_user, domain)):
             return False
```
Comment (gherceg) [inline corehq/apps/reports/dispatcher.py:312]: @biyeun or @mjriley, I want to use this specific permission `can_access_linked_domains` for this report. The better solution seems to be adding a specific report dispatcher that inherits from `DomainReportDispatcher`, but this also appears to be the precedent. Curious to hear any thoughts.

**Comment by gherceg** [comment]
Also want to flag [this line](https://github.com/dimagi/commcare-hq/pull/30077/files#diff-07657c6e5232742251036eb16896a92f0163cb70dbee4d0addd7f920cf3de913R312). I didn't make any changes here because this is the best way to avoid changing the existing feature flag workflow at the moment. I have added this as a task to address in the near future, but need to understand the access control solutions wants around linked projects.

File: corehq/apps/reports/dispatcher.py (hunk 2 above)
Comment (gherceg) [reply corehq/apps/reports/dispatcher.py]: Offlined with Biyeun and decided a new report dispatcher to house release management features is the way to go.

File: corehq/apps/reports/urls.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
 from .dispatcher import (
     CustomProjectReportDispatcher,
     ProjectReportDispatcher,
-    UserManagementReportDispatcher,
+    UserManagementReportDispatcher, ReleaseManagementReportDispatcher,
```
Comment (biyeun) [inline corehq/apps/reports/urls.py:24]: nit: new line

File: corehq/apps/reports/dispatcher.py [hunk 4]
```diff
@@ -297,12 +297,13 @@ def dispatch(self, request, *args, **kwargs):
...
         from corehq.motech.repeaters.views import DomainForwardingRepeatRecords
         from corehq.apps.export.views.incremental import IncrementalExportLogView
+
         from corehq.toggles import INCREMENTAL_EXPORTS
 
         if (report.endswith(DomainForwardingRepeatRecords.__name__)
                 and not domain_has_privilege(domain, privileges.DATA_FORWARDING)):
             return False
-        if (report.endswith(IncrementalExportLogView.__name__) and not INCREMENTAL_EXPORTS.enabled(domain)):
+        if report.endswith(IncrementalExportLogView.__name__) and not INCREMENTAL_EXPORTS.enabled(domain):
```
Comment (biyeun) [inline corehq/apps/reports/dispatcher.py:306]: curious why this change made it into this PR? seems unrelated

File: corehq/apps/linked_domain/util.py [hunk 5]
```diff
@@ -3,14 +3,29 @@
...
 from corehq.apps.app_manager.dbaccessors import get_app
 from corehq.apps.app_manager.exceptions import MultimediaMissingError
 from corehq.apps.hqmedia.models import CommCareMultimedia
 from corehq.apps.hqwebapp.tasks import send_html_email_async
 from corehq.apps.linked_domain.remote_accessors import fetch_remote_media
+from corehq.privileges import RELEASE_MANAGEMENT
 from corehq.util.timezones.conversions import ServerTime
 
 
+def can_access_linked_domains(user, domain):
```
Comment (biyeun) [inline corehq/apps/linked_domain/util.py:16]: new decorator(s), new tests? :)

would be nice to have tests for these directly

**Comment by biyeun** [comment]
more a higher-level thing... the PR references "tab" but from the image and the code, this is really adding ERM to the left-hand navigation for the Project Settings page, not creating a new tab. Feel like the name might be misleading at a glance.

File: corehq/apps/reports/dispatcher.py (hunk 4 above)
Comment (gherceg) [reply corehq/apps/reports/dispatcher.py:306]: Ah I originally had an additional if statement here, and PyCharm peer pressured me into removing redundant parenthesis. I'll put it back just to avoid confusion

**Comment by gherceg** [comment]
> more a higher-level thing... the PR references "tab" but from the image and the code, this is really adding ERM to the left-hand navigation for the Project Settings page, not creating a new tab. Feel like the name might be misleading at a glance.

Yeah I was basing that naming based on where these items live but makes sense that they aren't tabs. This also grew to include report privileges. I'll update the title.

File: corehq/apps/linked_domain/tests/test_decorators.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
+from unittest.mock import patch
+
+from django.test import SimpleTestCase
+
+from corehq.apps.linked_domain.util import can_access_linked_domains
+from corehq.util.test_utils import flag_enabled
+
+
+class TestCanAccessLinkedDomains(SimpleTestCase):
```
Comment (biyeun) [inline corehq/apps/linked_domain/tests/test_decorators.py:9]: yay! thanks

**Review by biyeun — APPROVED** [review]
